ansible_collections/arista/avd/roles/eos_validate_state/templates/generate_vars_for_testing.j2

```diff
@@ -1,23 +1,19 @@
-{# loopback0 reachability test #}
+{# Loopback reachability test #}
 {% set address = [] %}
 {% for node in groups[fabric_name] | arista.avd.natural_sort %}
-{%     if hostvars[node].type == 'l3leaf' %}
-{%         if hostvars[node].loopback_interfaces.Loopback0.ip_address is defined and hostvars[node].loopback_interfaces.Loopback0.ip_address is not none %}
-{%             do address.append(hostvars[node].loopback_interfaces.Loopback0.ip_address | ansible.netcommon.ipaddr('address')) %}
-{%         endif %}
+{%     if hostvars[node].router_bgp.peer_groups["EVPN-OVERLAY-PEERS"].update_source is arista.avd.defined %}
```

Comment:
Because you are accessing ```hostvars[node].router_bgp.peer_groups["EVPN-OVERLAY-PEERS"].update_source``` more than once, please set it to a temporary variable. Because accessing hostvars multiple times will increasing the run-time of the code.

Or alternatively set node_vars = hostvars[node] and then access different values using the temporary variable ```node_vars```. We used the similar technique in the recent refactor.